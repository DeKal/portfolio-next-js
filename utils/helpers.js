import { INIT_SELECTED_PAGE } from '~/consts/initState'
import $ from 'jquery'

export const initHashLocation = () => {
  if (location.hash) {
    const offset = $(location.hash).offset()
    window.scrollTo(offset.left, offset.top)
  }
}

export const isSafari = () => {
  var userAgent = navigator.userAgent.toLowerCase()
  if (userAgent.indexOf('safari') != -1) {
    // Safari agent will not have Chrome
    if (userAgent.indexOf('chrome') > -1) {
      return false
    } else {
      return true
    }
  }
}

export const setJqueryScrollEvent = () => {
  $('a').on('click', function(event) {
    // Make sure this.hash has a value before overriding default behavior
    if (this.hash) {
      // Prevent default anchor click behavior
      event.preventDefault()
      // Store hash
      var hash = this.hash

      $('html, body').animate(
        {
          scrollTop: $(hash).offset().top
        },
        400,
        function() {
          // Add hash (#) to URL when done scrolling (default click behavior)
          window.location.hash = hash
        }
      )
    } // End if
  })
}

export const cleanUpScrollEvent = () => {
  $('a').unbind()
}

export const getSelectedSection = () => {
  if (location.hash.length !== 0) {
    const selectedPage = location.hash.substr(1)
    return selectedPage
  }
  return INIT_SELECTED_PAGE
}

export const isChristmasDay = () => {
  const today = new Date()
  const month = today.getMonth() + 1
  const isDecember = month == 12
  const day = today.getDate()
  const isChristMasTime = 20 <= day <= 30

  return isDecember && isChristMasTime
}

export const isNewYearComing = () => {
  const today = new Date()
  const day = today.getDate()
  const isFirstTwoWeek = 1 <= day <= 16
  const month = today.getMonth() + 1
  const isJanuary = month == 1

  return isJanuary && isFirstTwoWeek
}
