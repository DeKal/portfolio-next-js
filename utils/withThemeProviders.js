import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import defaultTheme from '~/views/theme'

export const shallowWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

export const mountWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}
