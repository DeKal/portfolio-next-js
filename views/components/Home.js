import * as React from 'react'
import { HOME } from '~/consts/pages'
import styled from 'styled-components/macro'
import { Translate } from 'react-localize-redux'
import Container from 'react-bootstrap/Container'
import Anchor from '~/views/components/common/Anchor'
import Typist from '~/views/components/common/Typist'

const Home = () => (
  <HomeSection>
    <Anchor id={HOME} />
    <Background />
    <Overlay>
      <Container>
        <Header>
          <TitleName>
            <Translate id="Name" />
          </TitleName>
          <TitleJob>
            <Typist>
              <Translate id="Job" />
              <Translate id="Welcome" />
            </Typist>
          </TitleJob>
        </Header>
      </Container>
    </Overlay>
  </HomeSection>
)

const HomeSection = styled.section`
  position: relative;
  width: 100%;
  overflow: hidden;

  &::after {
    content: '';
    position: absolute;
    left: 0;
    bottom: 0;
    display: block;
    width: 100%;
    height: 0;
    clear: both;
    box-shadow: 0 0 2px 1px ${props => props.theme.colors.colorDark};
  }
`

const TitleName = styled.h1`
  margin-bottom: 10px;
  font-size: 100px;
  text-shadow: 1px 1px 1px ${props => props.theme.colors.colorDark};
  width: fit-content;
  background: ${props => props.theme.titleBgColor};

  @media screen and (max-width: 1200px) {
    font-size: 75px;
  }

  @media screen and (max-width: 800px) {
    font-size: 50px;
  }
`

const TitleJob = styled.h2`
  width: fit-content;
  background: ${props => props.theme.titleBgColor};
  color: ${props => props.theme.colors.colorLight};
  font-family: ${props => props.theme.fonts.fontTrebuchet};
  text-shadow: 1px 1px 2px ${props => props.theme.colors.colorDark};
`

const Background = styled.div`
  position: absolute;
  min-width: 100%;
  min-height: 100%;
  z-index: 0;
  background: transparent url(${props => props.theme.images.background}) center
    0 no-repeat;
  background-attachment: fixed;
  background-size: cover;
`

const Overlay = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
  background-color: ${props => props.theme.images.overlayBgColor};
  zoom: 1;
  background-image: url(${props => props.theme.images.overlay});
  background-repeat: repeat;
  background-attachment: fixed;
`

const Header = styled.header`
  padding: ${props => props.theme.headerPadding};
  text-align: left;

  @media screen and (max-width: 800px) {
    text-align: center;
  }

  @media screen and (max-width: 600px) {
    padding: 100px 0 300px;
  }
`
export default Home
