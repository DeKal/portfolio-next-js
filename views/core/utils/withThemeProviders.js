import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import homeTheme from '~/theme/theme'
import baseTheme from '~/theme/baseTheme'

/* istanbul ignore next */
export const shallowWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

/* istanbul ignore next */
export const shallowWithDivingTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children).dive()
}

/* istanbul ignore next */
export const mountWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}

/* istanbul ignore next */
export const shallowWithBaseTheme = (children, theme = baseTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}
