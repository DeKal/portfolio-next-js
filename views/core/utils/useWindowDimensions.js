import { useState, useEffect } from 'react'
import { getWindowDimensions } from '~/views/core/utils/helpers'

/* istanbul ignore next */
const useWindowDimensions = () => {
  const [windowDimensions, setWindowDimensions] = useState(
    getWindowDimensions()
  )

  useEffect(() => {
    /* istanbul ignore next */
    function handleResize() {
      setWindowDimensions(getWindowDimensions())
    }

    window.addEventListener('resize', handleResize)
    /* istanbul ignore next */
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  return windowDimensions
}

export default useWindowDimensions
