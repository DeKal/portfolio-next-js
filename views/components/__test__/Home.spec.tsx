import * as React from 'react'
import { shallowWithTheme } from '~/utils/withThemeProviders'
import Home from '~/views/components/Home'

describe('Home', () => {

  it('should render without throwing an error', function () {
    const wrap = shallowWithTheme(<Home />)
    expect(wrap).toMatchSnapshot()
  })

})