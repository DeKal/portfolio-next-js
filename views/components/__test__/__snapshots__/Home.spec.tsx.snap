// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render without throwing an error 1`] = `
<Home>
  <Home__HomeSection>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Home__HomeSection-sc-1eskvms-0",
            "isStatic": false,
            "lastClassName": "blNagb",
            "rules": Array [
              "position:relative;width:100%;overflow:hidden;&::after{content:'';position:absolute;left:0;bottom:0;display:block;width:100%;height:0;clear:both;box-shadow:0 0 2px 1px ",
              [Function],
              ";}",
            ],
          },
          "displayName": "Home__HomeSection",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Home__HomeSection-sc-1eskvms-0",
          "target": "section",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <section
        className="Home__HomeSection-sc-1eskvms-0 blNagb"
      >
        <Anchor
          id="Home"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Anchor-sc-31ub0h-0",
                  "isStatic": false,
                  "lastClassName": "idBzXx",
                  "rules": Array [
                    "position:absolute;top:",
                    [Function],
                    "px;",
                  ],
                },
                "displayName": "Anchor",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Anchor-sc-31ub0h-0",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            id="Home"
          >
            <span
              className="Anchor-sc-31ub0h-0 idBzXx"
              id="Home"
            />
          </StyledComponent>
        </Anchor>
        <Home__Background>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Home__Background-sc-1eskvms-3",
                  "isStatic": false,
                  "lastClassName": "eTDPSm",
                  "rules": Array [
                    "position:absolute;min-width:100%;min-height:100%;z-index:0;background:transparent url(",
                    [Function],
                    ") center 0 no-repeat;background-attachment:fixed;background-size:cover;",
                  ],
                },
                "displayName": "Home__Background",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Home__Background-sc-1eskvms-3",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="Home__Background-sc-1eskvms-3 eTDPSm"
            />
          </StyledComponent>
        </Home__Background>
        <Home__Overlay>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Home__Overlay-sc-1eskvms-4",
                  "isStatic": false,
                  "lastClassName": "bZmRRN",
                  "rules": Array [
                    "position:relative;width:100%;height:100%;background-color:",
                    [Function],
                    ";zoom:1;background-image:url(",
                    [Function],
                    ");background-repeat:repeat;background-attachment:fixed;",
                  ],
                },
                "displayName": "Home__Overlay",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Home__Overlay-sc-1eskvms-4",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="Home__Overlay-sc-1eskvms-4 bZmRRN"
            >
              <Container
                fluid={false}
              >
                <div
                  className="container"
                >
                  <Home__Header>
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "Home__Header-sc-1eskvms-5",
                            "isStatic": false,
                            "lastClassName": "goxKxF",
                            "rules": Array [
                              "padding:",
                              [Function],
                              ";text-align:left;@media screen and (max-width:800px){text-align:center;}@media screen and (max-width:600px){padding:100px 0 300px;}",
                            ],
                          },
                          "displayName": "Home__Header",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "Home__Header-sc-1eskvms-5",
                          "target": "header",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <header
                        className="Home__Header-sc-1eskvms-5 goxKxF"
                      >
                        <Home__TitleName>
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "Home__TitleName-sc-1eskvms-1",
                                  "isStatic": false,
                                  "lastClassName": "WdUdf",
                                  "rules": Array [
                                    "margin-bottom:10px;font-size:100px;text-shadow:1px 1px 1px ",
                                    [Function],
                                    ";width:fit-content;background:",
                                    [Function],
                                    ";@media screen and (max-width:1200px){font-size:75px;}@media screen and (max-width:800px){font-size:50px;}",
                                  ],
                                },
                                "displayName": "Home__TitleName",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "Home__TitleName-sc-1eskvms-1",
                                "target": "h1",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <h1
                              className="Home__TitleName-sc-1eskvms-1 WdUdf"
                            >
                              <Translate
                                id="Name"
                              >
                                <WrappedTranslate
                                  context={
                                    Object {
                                      "activeLanguage": undefined,
                                      "addTranslation": [Function],
                                      "addTranslationForLanguage": [Function],
                                      "defaultLanguage": undefined,
                                      "ignoreTranslateChildren": false,
                                      "initialize": [Function],
                                      "languages": Array [],
                                      "renderToStaticMarkup": false,
                                      "setActiveLanguage": [Function],
                                      "translate": [Function],
                                    }
                                  }
                                  id="Name"
                                >
                                  Missing translationId: Name for language: \${ languageCode }
                                </WrappedTranslate>
                              </Translate>
                            </h1>
                          </StyledComponent>
                        </Home__TitleName>
                        <Home__TitleJob>
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "Home__TitleJob-sc-1eskvms-2",
                                  "isStatic": false,
                                  "lastClassName": "WzpRR",
                                  "rules": Array [
                                    "width:fit-content;background:",
                                    [Function],
                                    ";color:",
                                    [Function],
                                    ";font-family:",
                                    [Function],
                                    ";text-shadow:1px 1px 2px ",
                                    [Function],
                                    ";",
                                  ],
                                },
                                "displayName": "Home__TitleJob",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "Home__TitleJob-sc-1eskvms-2",
                                "target": "h2",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <h2
                              className="Home__TitleJob-sc-1eskvms-2 WzpRR"
                            >
                              <Translate
                                id="Job"
                              >
                                <WrappedTranslate
                                  context={
                                    Object {
                                      "activeLanguage": undefined,
                                      "addTranslation": [Function],
                                      "addTranslationForLanguage": [Function],
                                      "defaultLanguage": undefined,
                                      "ignoreTranslateChildren": false,
                                      "initialize": [Function],
                                      "languages": Array [],
                                      "renderToStaticMarkup": false,
                                      "setActiveLanguage": [Function],
                                      "translate": [Function],
                                    }
                                  }
                                  id="Job"
                                >
                                  Missing translationId: Job for language: \${ languageCode }
                                </WrappedTranslate>
                              </Translate>
                            </h2>
                          </StyledComponent>
                        </Home__TitleJob>
                      </header>
                    </StyledComponent>
                  </Home__Header>
                </div>
              </Container>
            </div>
          </StyledComponent>
        </Home__Overlay>
      </section>
    </StyledComponent>
  </Home__HomeSection>
</Home>
`;
